### PR TITLE
feat: implement ekwidon

### PR DESCRIPTION
**Pull request by dmikalova**
> [!NOTE]
> **Medium Risk**
> Adds a large batch of new card scripts with complex interactions (control-changing, forging, conditional multi-step effects) that could surface edge-case engine bugs despite being mostly additive. Includes new destruction log coverage that may reveal or require tweaks to existing destroy/ward sequencing.
> 
> **Overview**
> Implements a batch of new `14-DM` Ekwidon card behaviors (play/fight/reap/action/reaction/persistent effects), including conditional and multi-step resolutions like control swaps (`even-swap`), key-cost modification during the opponent’s next turn (`oŏni-lars`), and forging with cost reduction based on counters/amber moved (`forged-currency`).
> 
> Adds comprehensive card-level test coverage for these new behaviors and edge cases (empty selections, optional prompts, changed control/ownership), plus a new `destroy` message test to validate chained destruction logging with ward prevention.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 51869476a435c3008639ad2d895abef55b28e130. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: server/game/cards/14-DM/BleaForh.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                    choices: {
+                        0: () => true,
+                        1: (context) =>
+                            context.targets.creature && context.targets.creature.damage >= 1,
+                        2: (context) =>
+                            context.targets.creature && context.targets.creature.damage >= 2
+                    }
+                }
+            },
+            effect: 'heal {1} for {2} damage',
```
Comment (strib) [inline server/game/cards/14-DM/BleaForh.js]: Should mention the exalting.

File: server/game/cards/14-DM/ConradFisique.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    // Enhance.
+    // Play: You may move each +1 power counter from a creature to another
+    // creature.
+    setupCardAbilities(ability) {
+        this.play({
+            optional: true,
+            targets: {
+                source: {
+                    cardType: 'creature',
+                    cardCondition: (card) => card.hasToken('power'),
```
Comment (strib) [inline server/game/cards/14-DM/ConradFisique.js]: You should be able to choose a creature with no power tokens.

File: server/game/cards/14-DM/EvenSwap.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            cardType: 'creature',
+            controller: 'opponent',
+            gameAction: ability.actions.cardLastingEffect((context) => ({
+                duration: 'lastingEffect',
+                effect: ability.effects.takeControl(context.player)
+            }))
+        });
+
+        this.play({
+            target: giveTarget('a'),
```
Comment (strib) [inline server/game/cards/14-DM/EvenSwap.js]: `giveTarget` doesn't take a param.  Same with the other call below and the `takeTarget` calls.

File: server/game/cards/14-DM/ForgedCurrency.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
+const Card = require('../../Card.js');
+
+class ForgedCurrency extends Card {
+    // Play: Move each +1 power counter and each A from friendly creatures
+    // to the common supply. Forge a key at +6 current cost, reduced by 1
+    // for the total number of +1 power counters and A moved to the common
+    // supply this way (to a minimum of 6).
+    setupCardAbilities(ability) {
+        this.play({
+            effect: 'move all +1 power counters and amber from friendly creatures to the common supply',
```
Comment (strib) [inline server/game/cards/14-DM/ForgedCurrency.js]: Should probably have the counts in this message, annoying as that is.

File: server/game/cards/14-DM/NowAndLater.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+                cardType: 'creature',
+                gameAction: ability.actions.returnToHand()
+            },
+            then: {
+                alwaysTriggers: true,
+                target: {
+                    activePromptTitle: 'Choose a card to archive',
+                    location: 'hand',
+                    controller: 'self',
+                    gameAction: ability.actions.archive()
```
Comment (strib) [inline server/game/cards/14-DM/NowAndLater.js:20]: Only the first action gets logged, this one and the later ones don't.

File: server/game/cards/14-DM/VastStoik.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    // Taunt.
+    // After Fight: Move 1A from a creature to the common supply. If you do,
+    // draw a card.
+    setupCardAbilities(ability) {
+        this.fight({
+            target: {
+                activePromptTitle: 'Choose a creature',
+                cardType: 'creature',
+                gameAction: ability.actions.conditional((context) => ({
+                    condition: context.target && context.target.hasToken('amber'),
```
Comment (strib) [inline server/game/cards/14-DM/VastStoik.js]: I think you can choose a creature with no amber on it, if you don't want to move any amber and one exists.

File: test/server/messages/destroy.spec.js [hunk 7]
```diff
@@ -73,4 +73,36 @@ describe('Destroy Messages', function () {
             ]);
         });
     });
+
+    describe('chained destroys from Harbinger of Doom with a warded creature', function () {
```
Comment (strib) [inline test/server/messages/destroy.spec.js:77]: Seems unrelated but sure.

**Review by strib — APPROVED** [review]

File: test/server/messages/destroy.spec.js (hunk 7 above)
Comment (dmikalova) [reply test/server/messages/destroy.spec.js:77]: I noticed this while running some other tests, and wanted to make sure the messaging was captured.

File: server/game/cards/14-DM/BleaForh.js [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+                cardType: 'creature',
+                gameAction: ability.actions.heal({ amount: 2, upTo: true })
+            },
+            preferActionPromptMessage: true,
+            then: (preThenContext) => ({
+                condition: (context) =>
+                    (context.preThenEvents || []).some(
+                        (event) => !event.cancelled && event.amount > 0
+                    ),
+                message: '{0} uses {1} to exalt {3} with {4} amber',
```
[the remark this answers is not in the document]
Comment (strib) [reply server/game/cards/14-DM/BleaForh.js:19]: Tested and it seems to work the way it is.
